src/main/java/org/nuxeo/segment/io/listener/SegmentIOAsyncListener.java

```diff
@@ -87,7 +87,7 @@ protected void processEvents(Map<String, List<SegmentIOMapper>> event2Mappers, E
                 Principal princ = event.getContext().getPrincipal();
                 SegmentIO service = Framework.getService(SegmentIO.class);
                 if (!service.mustTrackprincipal(princ.getName())) {
-                    return;
+                    continue;
```

Comment:
Shouldn't be `break;` here? As the principal is linked to the event, not the the mapper.